### PR TITLE
Mainly addresses #3 (https://github.com/bminer/intel-hex.js/issues/3)…

### PR DESCRIPTION
**Pull request by Scrandre**
…. Adds the optional starting address offset parameter (e.g. microcontroller images) and limits the size of the buffer with the value of bufferSize. Prevents out of memory issues on low RAM machines since it only creates the buffer of the required size from the expected range.

### CODE REVIEW

**Comment by bminer** [comment]
This is a really nice compromise that addresses #3. I will review shortly. Thank you for the PR!!!

File: index.js [hunk 1]
```diff
@@ -30,8 +31,12 @@ exports.parse = function parseIntelHex(data, bufferSize) {
 		highAddress = 0, //upper address
 		startSegmentAddress = null,
 		startLinearAddress = null,
+		addressOffset = 0 || addressOffset,
```
Comment (bminer) [inline index.js:34]: Should be `addressOffset ||= 0`

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "intel-hex",
-	"version": "0.1.2",
+	"version": "0.1.3",
```
Comment (bminer) [inline package.json]: Change to `0.2.0`

File: index.js [hunk 3]
```diff
@@ -30,8 +31,12 @@ exports.parse = function parseIntelHex(data, bufferSize) {
...
 		lineNum = 0, //Line number in the Intel Hex string
 		pos = 0; //Current position in the Intel Hex string
+
+	buf.fill(EMPTY_VALUE);
```
Comment (bminer) [inline index.js:38]: Why is this needed?  Can it be avoided?

**Review by bminer — CHANGES_REQUESTED** [review]
There should be more error handling to catch cases where the Intel Hex file specifies data before the `addressOffset` parameter.

Overall, I like the idea of adding the `addressOffset` parameter!

File: index.js (hunk 3 above)
Comment (Scrandre) [reply index.js:38]: Originally I didn't include it and the buffer's end was populated with 0s instead of FFs, so I pre-filled it with FFs to resolve, but maybe there is a better place to implement it

**Comment by Scrandre** [comment]
> There should be more error handling to catch cases where the Intel Hex file specifies data before the addressOffset parameter.

Well, the intent is actually to discard anything before the start of the target memory address, since although the HEX file includes data there, that data was irrelevant to the currently programmed part of the controller (but belongs to another part)
I guess the original HEX file comes like a bundle from which you extract only the parts that you need

**Comment by bminer** [comment]
Okay, yeah, that makes a lot of sense! That behavior should be explicitly stated in the README.

> I guess the original HEX file comes like a bundle from which you extract only the parts that you need

Yeah, I agree. Well said.

File: index.js (hunk 1 above)
Comment (Scrandre) [reply index.js:34]: That's a good point (although still rarely seen) but I have to use Node 12 and from what I understand this would bump the requirements to Node 15

**Comment by bminer** [comment]
Thank you for the PR!  It has inspired me to work on a similar PR #7.  Could you take a look at #7 to see if this also addresses the problem?
